homeassistant/components/victron_gx_mqtt/config_flow.py

```diff
@@ -0,0 +1,269 @@
+"""Config flow for victron mqtt integration."""
+
+# Future imports
+from __future__ import annotations
+
+# Standard library imports
+from collections.abc import Sequence
+import logging
+from types import MappingProxyType
+from typing import Any
+from urllib.parse import urlparse
+
+# Third-party imports
+from victron_mqtt import CannotConnectError, DeviceType, Hub as VictronVenusHub
+import voluptuous as vol
+
+# Home Assistant imports
+from homeassistant.config_entries import (
+    ConfigEntry,
+    ConfigFlow,
+    ConfigFlowResult,
+    OptionsFlow,
+)
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_USERNAME,
+)
+from homeassistant.helpers.selector import SelectOptionDict
+from homeassistant.helpers.service_info.ssdp import SsdpServiceInfo
+
+# Local application imports
+from .const import (
+    CONF_INSTALLATION_ID,
+    CONF_MODEL,
+    CONF_ROOT_TOPIC_PREFIX,
+    CONF_SERIAL,
+    CONF_SIMPLE_NAMING,
+    CONF_UPDATE_FREQUENCY_SECONDS,
+    DEFAULT_HOST,
+    DEFAULT_PORT,
+    DEFAULT_SIMPLE_NAMING,
+    DEFAULT_UPDATE_FREQUENCY_SECONDS,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEVICE_CODES: Sequence[SelectOptionDict] = [
+    {"value": device_type.code, "label": device_type.string}
+    for device_type in DeviceType
+    if device_type.string != "<Not used>"
+]
+
+
+def _get_user_schema(defaults: MappingProxyType[str, Any] | None = None) -> vol.Schema:
+    """Get the user data schema with optional defaults."""
+    if defaults is None:
+        defaults = MappingProxyType({})
+
+    return vol.Schema(
+        {
+            vol.Required(CONF_HOST, default=defaults.get(CONF_HOST, DEFAULT_HOST)): str,
+            vol.Required(CONF_PORT, default=defaults.get(CONF_PORT, DEFAULT_PORT)): int,
+            # Using suggested_value to be able to set empty string as default
+            vol.Optional(
+                CONF_USERNAME,
+                description={"suggested_value": f"{defaults.get(CONF_USERNAME, '')}"},
+            ): str,
+            vol.Optional(
+                CONF_PASSWORD,
+                description={"suggested_value": f"{defaults.get(CONF_PASSWORD, '')}"},
+            ): str,
+            vol.Required(CONF_SSL, default=defaults.get(CONF_SSL, False)): bool,
+            vol.Optional(
+                CONF_SIMPLE_NAMING,
+                default=defaults.get(CONF_SIMPLE_NAMING, DEFAULT_SIMPLE_NAMING),
+            ): bool,
+            vol.Optional(
+                CONF_ROOT_TOPIC_PREFIX,
+                description={
+                    "suggested_value": f"{defaults.get(CONF_ROOT_TOPIC_PREFIX, '')}"
+                },
+            ): str,
+            vol.Optional(
+                CONF_UPDATE_FREQUENCY_SECONDS,
+                default=defaults.get(
+                    CONF_UPDATE_FREQUENCY_SECONDS, DEFAULT_UPDATE_FREQUENCY_SECONDS
+                ),
+            ): int,
```

Comment:
What will this be used for? In general we don't allow configurable interval as people can use an action to update entities and thus can use an automation to trigger updates on their own interval